cluster: add projected volume support

plesoun-stein opened the pull request:
I added projected volume support in In cluster specification
tested. 
my simple testing env:
namespace: aaa
clusterName: firstcnpg-cluster
values: 
```
  projectedVolumeTemplate:
    sources:
      - secret:
          name: crl
          items:
            - key: crl.pem
              path: crl/crl-01.pem
```
secret successfully mounted in `/projected/crl/crl-01.pem`

plesoun-stein committed:
cluster: add projected volume support

Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

plesoun-stein committed:
added values.schema.json

Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

itay-grudev commented:
Could you please include an example to the [non-default configuration test](https://github.com/cloudnative-pg/charts/tree/main/charts/cluster/test/postgresql-cluster-configuration)?

itay-grudev committed:
Increased Scheduled Backups test cleanup timeout (#512)

* Increased Scheduled Backups test cleanup timeout

Currently the scheduled backups test times out causing a test failure, simply because it take a bit longer to remove the backup objects.

Signed-off-by: Itay Grudev <itay@verito.digital>

* Increased timeout to 5min

Signed-off-by: Itay Grudev <itay@verito.digital>

---------

Signed-off-by: Itay Grudev <itay@verito.digital>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

itay-grudev committed:
Running tests in parallel with a matrix strategy (#381)

* Running tests in parallel with a matrix strategy

---------

Signed-off-by: Itay Grudev <itay.grudev@essentim.com>
Signed-off-by: Itay Grudev <itay@verito.digital>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

kaisoz committed:
feat: Allow the user to set the operator update strategy (#450)

Signed-off-by: Tomas Tormo <tomas.tormo@nearform.com>
Co-authored-by: Itay Grudev <itay@verito.digital>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

renovate[bot] committed:
chore(deps): update sigstore/cosign-installer action to v3.8.0 (#513)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

renovate[bot] committed:
chore(deps): update actions/checkout action to v4.2.2 (#514)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

renovate[bot] committed:
chore(deps): update helm/kind-action action to v1.12.0 (#468)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

renovate[bot] committed:
chore(deps): update dependency ubuntu to v24 (#515)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

itay-grudev committed:
Bug Fix: Added extra check for inheritFromIAMRole when creating recovery secret (#503)

Signed-off-by: Itay Grudev <itay@verito.digital>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

itay-grudev committed:
Improved release documentation for the operator chart (#474)

Signed-off-by: Itay Grudev <itay@verito.digital>
Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

plesoun-stein committed:
cluster test: add projected volume

Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

plesoun-stein commented:
> Could you please include an example in the [non-default configuration test](https://github.com/cloudnative-pg/charts/tree/main/charts/cluster/test/postgresql-cluster-configuration)?

I have added projected volume definiton into values file `01-non_default_configuration_cluster.yaml`

```
  projectedVolumeTemplate:
    sources:
      - secret:
          name: ca-secret 
          items:
          - key: ca.crt
            path: crt/ca-01.pem
```

I have reused a `ca-secret` from above and I expect, there is a ca.crt key in the secret.

plesoun-stein committed:
cluster: values.yaml: fix trailing spaces

Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

plesoun-stein commented:
Ihave fixed trailing spaces error

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

itay-grudev commented:
The test here is missing an assertion.

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

mercuryyy commented:
Can this be merged ?

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
add assert for pojected volume

Signed-off-by: Tomas Steiner <plesoun.stein@gmail.com>

plesoun-stein commented:
> The test here is missing an assertion.

I have added assertion in `charts/cluster/test/postgresql-cluster-configuration/01-non_default_configuration_cluster-assert.yaml`
I hope it's on the right place.

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

pha91 commented:
@itay-grudev we're also very interested in this feature. Do we have to fix something or is it just forgotten in the hustle?

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

plesoun-stein committed:
Merge branch 'cloudnative-pg:main' into projected-volumes

itay-grudev commented:
Rebased and merged in #986
